pkg/cmd/unban_test.go

```diff
@@ -0,0 +1,144 @@
+package cmd_test
+
+import (
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/hash"
+	"github.com/kubesaw/ksctl/pkg/cmd"
+	"github.com/kubesaw/ksctl/pkg/configuration"
+	clicontext "github.com/kubesaw/ksctl/pkg/context"
+	"github.com/kubesaw/ksctl/pkg/ioutils"
+	. "github.com/kubesaw/ksctl/pkg/test"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestUnbanCommand(t *testing.T) {
+	t.Run("fails with no parameters", func(t *testing.T) {
+		cmd := cmd.NewUnbanCommand()
+		_, err := cmd.ExecuteC()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "accepts 1 arg(s), received 0")
+	})
+	t.Run("fails with more than 1 parameter", func(t *testing.T) {
+		cmd := cmd.NewUnbanCommand()
+		cmd.SetArgs([]string{"a", "b"})
+		_, err := cmd.ExecuteC()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "accepts 1 arg(s), received 2")
+	})
+	t.Run("runs with exactly 1 parameter", func(t *testing.T) {
+		cmd := cmd.NewUnbanCommand()
+		cmd.SetArgs([]string{"me@home"})
+		commandRan := false
+		cmd.RunE = func(cmd *cobra.Command, args []string) error {
+			require.Len(t, args, 1)
+			assert.Equal(t, "me@home", args[0])
+			commandRan = true
+			return nil
+		}
+		_, err := cmd.ExecuteC()
+		require.NoError(t, err)
+		assert.True(t, commandRan)
+	})
+}
+
+func TestUnbanWhenNoneExists(t *testing.T) {
+	// given
+	SetFileConfig(t, Host())
+	term := NewFakeTerminal()
+	newClient, _ := NewFakeClients(t)
+	ctx := clicontext.NewCommandContext(term, newClient)
+
+	// when
+	err := cmd.Unban(ctx, "me@home")
+
+	// then
+	require.NoError(t, err)
+	output := term.Output()
+	assert.NotContains(t, output, "User successfully unbanned")
+	assert.Contains(t, output, "No BannedUser objects found with given email.")
```

Comment:
Couldn't we do something like this?
```suggestion
        assert.Equal(t, "No BannedUser objects found with given email.\n", output)
```

Reply:
The command prints a lot of output on top of what we add, by having those 2 asserts, I'm discriminating between the unexpected success case and the expected failure case.

Reply:
I was able to find the reason for the superfluous output from the tests, so now the tests in `unban_test.go` use `Equal` on the output as you suggested. It was actually caused by the fact that a completely unrelated test in `logs_test.go` was setting a global variable `configuration.Verbose` (causing the verbose output) and was not resetting it back afte the test was done.